common/util/file_util.cc

```diff
@@ -46,6 +48,34 @@ absl::string_view Stem(absl::string_view filename) {
              : filename.substr(0, last_dot_pos);
 }
 
+bool UpwardFileSearch(std::string start, std::string filename,
+                      std::string& result) {
+  /* Convert to absolute path */
+  char absolute_path[PATH_MAX];
+  if (realpath(std::string(start).c_str(), absolute_path) < 0) {
+    return false;
+  }
+  start = std::string(absolute_path);
+
+  std::vector<absl::string_view> search = absl::StrSplit(start, '/');
+
+  /* Push back linter configuration file name */
+  search.push_back(filename);
+
+  while (search.size() > 1) {
+    std::string candidate = absl::StrJoin(search,  "/");
```

Comment:
Just an implementation idea that reduces string-copying and allocation:

* full path consists of a directory component and file (base) name component.
* maintain a `string_view` of the directory component
* find all (iterator) positions that contain `/`
* shrink the string_view from the back with `remove_suffix`
* `absl::StrCat` the shrinking directory component with file name -> candidate.

Reply:
Changed the implementation to somewhat reflect your suggestions. Is the current implementation acceptable?